README

```diff
@@ -274,3 +274,68 @@ refresh			Attempts to reinsert all modules that are currently
+
+Each process or resource (file, dir etc.) has a label (type) that describes what can be accessed.
+
+SELinux considers subject-access-object rule set. On a Linux system, subjects are processes, and object is the resource on which an action applies.
+
+Refpolicy contains the following file types:
+1. **te** - the file contains rules describing what the domain can do (labels of the subjects or objects). This file does not say that this process can run another file, it just says that the process labeled A can read the file labeled B.
+2. **fc** - describes the security context (labels) on a file, directory, etc., which will be applied when the policy is install. By default, directories or files inherit the context of their parents, but you can manually set the desired context through this file
+3. **if** - the interface file creates the macros that other modules will use to gain access to my resources.
+
+SELinux can run in one of three modes: **disabled**, **permissive**, or **enforcing**.
+Using the **disabled** mode means that no rules from the SELinux policy are applied and your system is not protected. Therefore, the disabled mode is not recommended.
+In the **permissive** mode, SELinux is active, the security policy is loaded, the file system is labeled and access denial entries are logged. However, the policy is not enforced and thus no access is actually denied.
+In the **enforced** mode, the security policy is applied. Each access that is not explicitly allowed by the policy is denied.
+
+To be able to check SELinux for possible access denials, need to set the **permissive** mode in the **meta-aos-vm** for **refpolicy-aos** recipe to the **DEFAULT_ENFORCING** variable. This will make it possible to use SELinux, but in logging mode.
+
+To diagnose denials security policies, you can use the following command `journalctl | grep avc`, if the log contains **avc: denied** that means it is an SELinux policy denial. This log will include a line like this:
+
+```
+AVC avc:  denied  { write } for  pid=1248 comm="dnsmasq" name="dev-log" dev="tmpfs" ino=20945 scontext=system_u:system_r:dnsmasq_t:s0 tcontext=system_u:object_r:syslogd_runtime_t:s0 tclass=sock_file permissive=0
+```
+
+- **scontext** - SELinux context of the process (source) that attempted the denied action.
+- **tcontext** -SELinux context of the object (target) the process attempted to access
+- **comm** - name of the command that was used to access the target context
+name
+- **tclass** - target object class
+- **dev** - device on which the target resides
+- **name** - name of the target
+- can contained **path** -  path to the object (target) the process attempted to access.
+
+Next we need to analyze the received log and find the corresponding **te** file by **scontext**, for example, for **dnsmasq_t** there is a corresponding **dnsmasq.te** file and in this file you need to add a rule that allows **scontext** to access **tcontext**.
```

Comment:
The next step is to analyze ...  **dnsmasq.te** file and in this file  rule that allows **scontext** to access **tcontext** should be added.

Reply:
Done